### PR TITLE
Update TPU CI requirement for Pull Requests

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Update TPU CI requirement for Pull Requests

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by awaelchli** [comment]
I also tried here #7212 the same ^^
It looks like this must be set in Github admin not in our config, correct @Borda

File: .github/workflows/ci_test-tpu.yml [hunk 1]
```diff
@@ -3,10 +3,8 @@ name: TPU tests
 on:
   push:
     branches: [master, "release/*"]
-# TODO: temporal disable TPU testing until we find way how to pass credentials to forked PRs
-#  pull_request:
-#    branches:
-#      - master
+  pull_request:
+    branches: [master, "release/*"]
```
Comment (Borda) [inline .github/workflows/ci_test-tpu.yml:7]: this won't work for forkerd PRs

**Review by Borda — CHANGES_REQUESTED** [review]

**Comment by Borda** [comment]
> Update TPU CI requirement for Pull Requests

where it the req update? or what the req means? required tests to pass on merge?

File: .github/workflows/ci_test-tpu.yml (hunk 1 above)
Comment (awaelchli) [reply .github/workflows/ci_test-tpu.yml:7]: we didn't know it has to be changed in the GitHub settings

**Comment by Borda** [comment]
according our conversation, this is not needed, but feel free to reopen or needed :]

**Pull request closed without merging**